wiki/database.md

```diff
@@ -29,20 +29,116 @@ The structure of the DB is the following:
 
 ### Drinks
 
-**OUTDATED**
+A drink is represented by the following JSON: 
 
 ```JSON
 {
-    "drink_name": "Expresso",
-    "drink_price": 4.0,
-    "drink_description": "An expresso coffee",
+    "drink_name": "Name of the coffee recipe or Custom Drink for custom made drinks",
+    "drink_price": "4.0 // nu cred ca ne trebuie",
+    "drink_description": "The description of the drink", 
     "preparation": {
-        "coffee_mg": 10,
-        "milk_mg": 12,
-        "water_mg": 22,
-        "sugar_mg": 11,
+        "coffee_mg": "10 -> coffee quantity",
+        "milk_mg": "12 -> milk quantity",
+        "water_mg": "22 -> water quanity",
+        "sugar_mg": "11 -> sugar quantity",
+        "milk_foam": "true -> with or without milk foam",
     },
 }
 ```
 
-### TBD
\ No newline at end of file
+### Recipes Table
+
+This table keeps the drink recipes. Each recipe is a drink with a particular name and preparation.
+
+Example of recipes:
+
+```JSON
+{
+    "drink_name": "Espresso", 
+    "drink_description": "An espresso coffee", 
+    "preparation": {
+        "coffee_mg": 7,
+        "milk_mg": 0,
+        "water_mg": 30,
+        "sugar_mg": 0,
+        "milk_foam": false,
+    },
+}
+```
+
+```JSON
+{
+    "drink_name": "Cappuccino", 
+    "drink_description": "A cappuccino coffee",
+    "preparation": {
+        "coffee_mg": 7,
+        "milk_mg": 160,
+        "water_mg": 30,
+        "sugar_mg": 0,
+        "milk_foam": true,
+    },
+}
+```
+
+```JSON
+{
+    "drink_name": "Caffe latte", 
+    "drink_description": "A latte coffee",
+    "preparation": {
+        "coffee_mg": 7,
+        "milk_mg": 250,
+        "water_mg": 30,
+        "sugar_mg": 20,
+        "milk_foam": true,
+    },
+}
+```
+
+```JSON
+{
+    "drink_name": "Flat White", 
+    "drink_description": "A flat white coffee",
+    "preparation": {
+        "coffee_mg": 14,
+        "milk_mg": 130,
+        "water_mg": 60,
+        "sugar_mg": 0,
+        "milk_foam": false,
+    },
+}
+```
+
+### CoffeeMachines Table
+
+This table keeps the information for each coffee machine.
+
+```JSON
+{
+    "machine_name": "Name of the machine", 
+    "location": "Location of the machine",
+    "status": "WORKING / OUT_OF_SERVICE / NEEDS_RESTOCK / ERROR"
```

Comment:
Vrei sa salvezi starea instantanee a unei masini de cafea in baza de date?
Mi se pare ca cel mai simplu ar fi sa nu tinem minte in baza de date nimic legat de masinile de cafea, si singurul care stie de ele este serverul, care se aboneaza pe topicurile unde masinile de cafea dau broadcast la starea lor.

Reply:
Da, dar cred ca ar trebui cumva sa stim si locatia masinii care ne da informatia. De exemplu, adminul vrea sa vada starea fiecarei masini intr-un panel si daca una dintre ele s-a stricat / trebuie facut restock, sa stie care e. 

O varianta ar fi ca masinile sa trimita JSON-ul asta prin broadcast.